onnx/defs/tensor/defs.cc

```diff
@@ -2206,77 +2214,238 @@ ONNX_OPERATOR_SET_SCHEMA(
+
+          // 'indices', 'inverse_indices', and 'counts' are 1-D tensors of
+          // unknown dimension.
           auto num_outputs = ctx.getNumOutputs();
           if (num_outputs >= 2) {
             indicesTensorProto = ctx.getOutputType(1);
             updateOutputElemType(ctx, 1, TensorProto::INT64);
-            indicesTensorProto->mutable_tensor_type()->mutable_shape()->add_dim();
+            indicesTensorProto->mutable_tensor_type()
+                ->mutable_shape()
+                ->add_dim();
           }
 
           if (num_outputs >= 3) {
             inverseIndicesTensorProto = ctx.getOutputType(2);
             updateOutputElemType(ctx, 2, TensorProto::INT64);
-            inverseIndicesTensorProto->mutable_tensor_type()->mutable_shape()->add_dim();
+            inverseIndicesTensorProto->mutable_tensor_type()
+                ->mutable_shape()
+                ->add_dim();
           }
 
           if (num_outputs >= 4) {
             countsTensorProto = ctx.getOutputType(3);
             updateOutputElemType(ctx, 3, TensorProto::INT64);
-            countsTensorProto->mutable_tensor_type()->mutable_shape()->add_dim();
+            countsTensorProto->mutable_tensor_type()
+                ->mutable_shape()
+                ->add_dim();
           }
 
           auto axisAttr = ctx.getAttribute("axis");
           if (!axisAttr) {
             // 'axis' is not provided. Input 'X' is flattened.
             // 'Y' is a 1-D tensor of unknown dimension.
             yTensorProto->mutable_tensor_type()->mutable_shape()->add_dim();
-          }
-          else 
-          {
+          } else {
             // 'axis' is provided.
             int axis = static_cast<int>(axisAttr->i());
 
-            // 'Y' has the same shape as 'X' except in the 'axis' dimension which is unknown.
-            const TensorShapeProto& input_shape = xTensorProto->tensor_type().shape();
+            // 'Y' has the same shape as 'X' except in the 'axis' dimension
+            // which is unknown.
+            const TensorShapeProto& input_shape =
+                xTensorProto->tensor_type().shape();
 
             for (int i = 0; i < input_shape.dim_size(); i++) {
-              auto* dim = yTensorProto->mutable_tensor_type()->mutable_shape()->add_dim();
-              if (i != axis){
+              auto* dim = yTensorProto->mutable_tensor_type()
+                              ->mutable_shape()
+                              ->add_dim();
+              if (i != axis) {
                 *dim = input_shape.dim(i);
               }
             }
           }
-      }));
+        }));
+
+static const char* GatherND_ver11_doc = R"DOC(
+Given `data` tensor of rank `r` >= 1, and `indices` tensor of rank `q` >= 1, this operator gathers 
+slices of `data` into an output tensor of rank `q + r - indices_shape[-1] - 1`.
+
+`indices` is an q-dimensional integer tensor, best thought of as a `(q-1)`-dimensional tensor of indices into `data`, 
```

Comment:
May be change "tensor of indices" to "tensor of index-tuples"?

Reply:
sure